manifests/server.pp

```diff
@@ -56,7 +56,9 @@
 ) {
   include profile_backup::common
 
-  if ($gid) and ($uid) {
+  if empty($gid) or empty($uid) {
+    fail('You must provide both gid and uid.')
+  } else {
```

Comment:
I don't the else is needed here (the call to `fail` should stop catalog compilation). Just close the if and the rest of the code can be outside of any flow control construct.

Reply:
I've removed the `else` and confirmed it works as expected.